src/hotspot/share/opto/regmask.hpp

```diff
@@ -64,28 +75,300 @@ class RegMask {
 
   static const unsigned int _WordBitMask = BitsPerWord - 1U;
   static const unsigned int _LogWordBits = LogBitsPerWord;
+
+  // RM_SIZE, but in number of machine words
   static const unsigned int _RM_SIZE     = LP64_ONLY(RM_SIZE >> 1) NOT_LP64(RM_SIZE);
+
+  // RM_SIZE_MIN, but in number of machine words
+  static const unsigned int _RM_SIZE_MIN =
+      LP64_ONLY(((RM_SIZE_MIN + 1) & ~1) >> 1) NOT_LP64(RM_SIZE_MIN);
+
+  // The last index (in machine words) of the (static) array of register mask
+  // bits
   static const unsigned int _RM_MAX      = _RM_SIZE - 1U;
 
+  // Compute a best-effort (statically known) upper bound for register mask
+  // size in 32-bit words. When extending/growing register masks, we should
+  // never grow past this size.
+  static const unsigned int RM_SIZE_MAX =
+      (((RM_SIZE_MIN << 5) +                // Slots for machine registers
+        (max_method_parameter_length * 2) + // Slots for incoming arguments
+        (max_method_parameter_length * 2) + // Slots for outgoing arguments
```

Comment:
What's the meaning of incoming vs outgoing arguments? Like this?
```
Incoming = from caller (outer nesting)
Outgoing = to nested call (inner nesting)
```

Reply:
Yes, you are correct. There is a detailed explanation in `x86_64.ad` ("Definition of frame structure and management information").